packages/core/core-flows/src/order/workflows/create-fulfillment.ts

```diff
@@ -157,6 +154,15 @@ function prepareInventoryUpdate({ reservations, order, input }) {
 
   for (const item of order.items) {
     const reservation = reservationMap[item.id]
+    if (!reservation) {
+      if (item.manage_inventory) {
+        throw new Error(
+          `No stock reservation found for item ${item.id} - ${item.title} (${item.variant_title})`
+        )
+      }
+      continue
+    }
+
```

Comment:
Should we refactor it to
```suggestion
      if (item.manage_inventory) {
       continue
      }
      
      const reservation = reservationMap[item.id]
    if (!reservation) {
        throw new Error(
          `No stock reservation found for item ${item.id} - ${item.title} (${item.variant_title})`
        )
    }

```

Reply:
That would change the behavior. Custom items won't have "manage_inventory" but they can or cannot have a reservation.
and adding extra `&&` or `||` to the `if` wouldn't make it easier to read.